### PR TITLE
Prefer the jvm directory for multiplatform builds

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: src/main/java/com/squareup/osstrich/JavadocPublisher.java [hunk 1]
```diff
@@ -161,17 +158,26 @@ private String findRelativePath(File directory, String artifactId) {
...
-              + Arrays.toString(Objects.requireNonNull(directory.listFiles())));
     }
+
+    // Look for subdirectories of the same name as the artifact and go a level deeper
+    File artifactDir = new File(directory, artifactId);
+    if (artifactDir.isDirectory()) {
+      return directory.getName() + File.separator + findRelativePath(artifactDir, artifactId);
+    }
+
+    // For Kotlin multiplatform look for a subdirectory named 'jvm' and go a level deeper.
```
Comment (JakeWharton) [inline src/main/java/com/squareup/osstrich/JavadocPublisher.java]: There also could be both jvm and android-specific platform modules.

Comment (swankjesse) [reply src/main/java/com/squareup/osstrich/JavadocPublisher.java]: Oooh tricky. Recommendations?

**Comment by JakeWharton** [comment]
Not really. Projects can be named anything

**Comment by swankjesse** [comment]
I think we land this and then support other directories as we build use cases for them. Perhaps with new commandline flags.

File: src/main/java/com/squareup/osstrich/JavadocPublisher.java [hunk 2]
```diff
@@ -157,21 +158,37 @@ private void writeIndexFiles(String groupId, Multimap<String, Artifact> artifact
...
+    File jvmDir = new File(directory, "jvm");
+    if (jvmDir.isDirectory()) {
+      String relativePath = findRelativePath(jvmDir, artifactId);
+      return relativePath != null
+          ? (directory.getName() + File.separator + relativePath)
+          : null;
+    }
+
+    log.error("Could not find a valid indexed path for " + artifactId
+        + ". Files are " + Arrays.toString(Objects.requireNonNull(directory.listFiles())));
```
Comment (hzsweers) [inline src/main/java/com/squareup/osstrich/JavadocPublisher.java]: readability nit - static import the `requireNonNull`

Comment (swankjesse) [reply src/main/java/com/squareup/osstrich/JavadocPublisher.java]: Fixed!

**Pull request merged**